mobius-core/src/test/java/com/spotify/mobius/MobiusLoopTest.java

```diff
@@ -440,20 +440,131 @@ public void eventsFromEffectHandlerDuringDisposeAreIgnored() throws Exception {
     assertFalse(updateWasCalled.get());
   }
 
+  @Test
+  public void disposingLoopWhileInitIsRunningDoesNotEmitNewState() throws Exception {
+    // Model changes emitted from the init function during dispose should be ignored.
+
+    // This test will start a loop and wait until (using the initRequested semaphore) the runnable
+    // that runs Init is posted to the event runner. The init function will then be blocked using
+    // the initLock semaphore. At this point, we proceed to add the observer then dispose of the
+    // loop. The loop is setup with an event source that returns a disposable that will unlock
+    // init when it is disposed. So when we dispose of the loop, that will unblock init as part of
+    // the disposal procedure. The test then waits until the init runnable has completed running.
+    // Completion of the init runnable means:
+    // a) init has returned a First
+    // b) that first has been unpacked and the model has been set on the store
+    // c) that model has been passed back to the loop to be emitted to any state observers
+    // Since we're in the process of disposing of the loop, we should see no states in our observer
```

Comment:
@pettermahlen I was able to simplify this test I think. Instead of blocking both init and dispose etc, I only blocked init, then had dispose unblock it, which would have the effect that we want for the test. The above comment describes how the test works